### ISSUE TITLE
viewForSupplementaryElementOfKind and referenceSizeForFooterInSection not called? 

### ISSUE THREAD

**Issue by tnahvu** [chiahsien/CHTCollectionViewWaterfallLayout#113]
Hi

First, thanks for the awesome library!  I'm not quite sure how to properly implement footer/header view.  Does the normal way not work?  Is there something else I need to do ?  viewForSupplementaryElementOfKind and referenceSizeForFooterInSection not called when using CHTCollectionViewWaterfallLayout.

Again, many thanks!

**Comment by chiahsien** [comment]
1. `viewForSupplementaryElementOfKind` will get called, you can see the demo for more information.
2. `referenceSizeForFooterInSection` will **NOT** get called, because it belongs to `UICollectionViewDelegateFlowLayout` but my `CHTCollectionViewDelegateWaterfallLayout` confirms to `UICollectionViewDelegate`.

**Comment by Gman9855** [comment]
@chiahsien I get a crash in `viewForSupplementaryElementOfKind` when `dequeueReusableSupplementaryViewOfKind:reuseIdentifier` is called.  

`Assertion failure in -[UICollectionViewData layoutAttributesForSupplementaryElementOfKind:atIndexPath:]`

I'm setting the footer height property of the layout like you say to do.  What am i missing here?  By the way I am changing between this layout and another layout which shows my cells in a tableview style layout, and when I am in that layout my footer view shows up just fine.

**Comment by cheriaa** [comment]
@Gman9855 Have you figured out the problem, cuz I'm facing exactly the same problem.

**Comment by Gman9855** [comment]
@cheriaa I've gotten it to at least not crash now.  You have to register the UICollectionReusableView class first.  So now what I have is the extra space I provided to the layout's footerHeight property, but the supplementary view doesn't appear, even though `viewForSupplementaryElementOfKind` gets called.

**Comment by f474283547** [comment]
hi, need help.
I custom UICollectionViewFlowLayout then viewForSupplementaryElementOfKind not called, how to solve ?

**Comment by fiancrs** [comment]
hi,'ve got a same problem.
when i make a UICollectionViewFlowLayout's subclass,and registered that into collectionView.But the method "viewForSupplementaryElementOfKind" could not be called.Contrarily, i had done this like 
`[[UICollectionView alloc] initWithFrame:CGRectZero collectionViewLayout:_layout]`,that method will be called.
Could you get my mind and help me to solve it?
Thanks

**Comment by f474283547** [comment]
of course,  you can reference (https://github.com/lqcjdx/YLTagsChooser) and http://www.jianshu.com/p/2686f7066ee2#


------------------ 原始邮件 ------------------
发件人: "fiancrs";<notifications@github.com>;
发送时间: 2016年11月18日(星期五) 中午11:18
收件人: "chiahsien/CHTCollectionViewWaterfallLayout"<CHTCollectionViewWaterfallLayout@noreply.github.com>; 
抄送: "fanchongxi"<474283547@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [chiahsien/CHTCollectionViewWaterfallLayout]viewForSupplementaryElementOfKind and referenceSizeForFooterInSection notcalled? (#113)




hi,'ve got a same problem.
 when i make a UICollectionViewFlowLayout's subclass,and registered that into collectionView.But the method "viewForSupplementaryElementOfKind" could not be called.Contrarily, i had done this like 
 [[UICollectionView alloc] initWithFrame:CGRectZero collectionViewLayout:_layout],that method will be called.
 Could you get my mind and help me to solve it?
 Thanks

**Comment by fiancrs** [comment]
@f474283547 不愧是自己人，效率可以，感谢

**Comment by f474283547** [comment]
哈哈 ,客气~


------------------ 原始邮件 ------------------
发件人: "fiancrs";<notifications@github.com>;
发送时间: 2016年11月18日(星期五) 中午11:29
收件人: "chiahsien/CHTCollectionViewWaterfallLayout"<CHTCollectionViewWaterfallLayout@noreply.github.com>; 
抄送: "fanchongxi"<474283547@qq.com>; "Mention"<mention@noreply.github.com>; 
主题: Re: [chiahsien/CHTCollectionViewWaterfallLayout]viewForSupplementaryElementOfKind and referenceSizeForFooterInSection notcalled? (#113)




@f474283547 不愧是自己人，效率可以，感谢

**Comment by AAChartModel** [comment]
Same issue, Has the problem been resolved?
